srcs/Context/CommandsHeaders.hpp

```diff
@@ -0,0 +1,8 @@
+#ifndef COMMANDSHEADERS_HPP
```

Comment:
I understand that you have a lot of includes in vscode, but in Clion we don't have such problem, all includes just shrink to one. Personally, I think that, include for include confuses.